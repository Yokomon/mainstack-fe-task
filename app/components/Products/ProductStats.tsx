"use client";

import { FiInfo } from "@react-icons/all-files/fi/FiInfo";

interface ProductStatsProps {
  amount: number;
  title: string;
}

const ProductStats: React.FC<ProductStatsProps> = ({ title, amount }) => {
  return (
    <div className="p-2">
      <div className="flex items-center tracking-tight font-medium justify-between text-sm text-gray-500">
        <h3>{title}</h3>
        <FiInfo className="cursor-pointer text-gray-400" size={20} />
      </div>
      <div className="text-2xl font-bold tracking-wide py-2">
        <h1>USD {amount}</h1>
      </div>
    </div>
  );
};

export default ProductStats;
