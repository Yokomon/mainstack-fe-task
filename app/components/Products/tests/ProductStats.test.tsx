import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import ProductStats from "../ProductStats";

const mockProductStatsData = {
  title: "Total Revenue",
  amount: 1500,
};

describe("ProductStats Component", () => {
  it("renders without crashing", () => {
    render(
      <ProductStats
        title={mockProductStatsData.title}
        amount={mockProductStatsData.amount}
      />
    );
    expect(screen.getByText("Total Revenue")).toBeInTheDocument();
  });

  it("displays the correct title", () => {
    render(
      <ProductStats
        title={mockProductStatsData.title}
        amount={mockProductStatsData.amount}
      />
    );
    expect(screen.getByText("Total Revenue")).toBeInTheDocument();
  });

  it("displays the correct amount", () => {
    render(
      <ProductStats
        title={mockProductStatsData.title}
        amount={mockProductStatsData.amount}
      />
    );
    expect(screen.getByText("USD 1500")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <ProductStats
        title={mockProductStatsData.title}
        amount={mockProductStatsData.amount}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
