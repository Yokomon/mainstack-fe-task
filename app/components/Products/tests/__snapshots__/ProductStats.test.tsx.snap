// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductStats Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-2"
  >
    <div
      class="flex items-center justify-between text-sm text-gray-500"
    >
      <h3>
        Total Revenue
      </h3>
      <svg
        class="cursor-pointer text-gray-400"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="8"
          y2="8"
        />
      </svg>
    </div>
    <div
      class="text-2xl font-bold tracking-wide py-2"
    >
      <h1>
        USD 1500
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
